docs/source/extension/plugin_system.rst

```diff
@@ -0,0 +1,179 @@
+.. Copyright (c) Jupyter Development Team.
+.. Distributed under the terms of the Modified BSD License.
+
+The JupyterLab Plugin System
```

Comment:
Could this be possibly moved out into a separate pull request? Two reasons:
1. We could more quickly merge the dual compatibility doc delivering the excellent piece above to the users/developers
2. This document would receive more attention from a wider group of developers, because there are folks who review pull request based on the title, I frankly did not expect to find this document in a PR titled ` Open
Lab/NB7 extension compatibility docs` :)

Reply:
The plugin system doc has been split out [here](https://github.com/jupyterlab/jupyterlab/pull/15417).